### PR TITLE
Add general analyzer manager tests

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Review by Or-Geva — CHANGES_REQUESTED** [review]

File: src/main/scanLogic/scanRunners/binaryRunner.ts [hunk 1]
```diff
@@ -113,40 +121,38 @@ export abstract class BinaryRunner {
...
+                binaryVars[BinaryRunner.ENV_PASSWORD] = this._connectionManager.password;
             }
-            // Proxy information - environment variable
+            // Optional proxy information - environment variable
             let proxyHttpUrl: string | undefined = process.env['HTTP_PROXY'];
             let proxyHttpsUrl: string | undefined = process.env['HTTPS_PROXY'];
-            // Proxy information - vscode configuration override
+            // Optional proxy information - vscode configuration override
             let optional: IProxyConfig | boolean = ConnectionUtils.getProxyConfig();
             if (optional) {
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/main/scanLogic/scanRunners/binaryRunner.ts]: we first take the environment variable is they exists and store them in proxyHttpUrl/proxyHttpsUrl.
if the proxy settings in vscode is defined it takes priorty and override the value in variable.
all the changes and checks are preformed on the local variable, if both sources not exists the local variable will remain undefined and we will not change binaryVars at all

File: src/main/scanLogic/scanRunners/binaryRunner.ts [hunk 2]
```diff
@@ -113,40 +121,38 @@ export abstract class BinaryRunner {
      * @param executionLogDirectory - the directory that the log will be written into, if not provided the log will be written in stdout/stderr
      * @returns list of environment variables to use while executing the runner or unidentified if credential not set
      */
-    private createEnvForRun(executionLogDirectory?: string): NodeJS.ProcessEnv | undefined {
+    public createEnvForRun(executionLogDirectory?: string): NodeJS.ProcessEnv | undefined {
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/main/scanLogic/scanRunners/binaryRunner.ts:124]: I don't think so, this is per request argument, like directory it changes each scan. for each run we create temporary folder to serve as an environment for the binary to run inside, this is the value that is passed to it and the destination that the log will be created inside

File: src/main/scanLogic/scanRunners/applicabilityScan.ts [hunk 3]
```diff
@@ -63,12 +65,12 @@ export class ApplicabilityRunner extends BinaryRunner {
...
     ): Promise<ApplicabilityScanResponse> {
         let request: ApplicabilityScanRequest = {
-            type: 'analyze-applicability',
+            type: ApplicabilityRunner.NAME,
             roots: [directory],
             cve_whitelist: Array.from(cveToRun),
             skipped_folders: skipFolders
         } as ApplicabilityScanRequest;
-        return this.run(checkCancel, false, request).then(response => this.generateResponse(response?.runs[0]));
+        return this.run(checkCancel, request).then(response => this.generateResponse(response?.runs[0]));
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/main/scanLogic/scanRunners/applicabilityScan.ts]: should wait, thanks

File: src/main/scanLogic/scanRunners/binaryRunner.ts [hunk 4]
```diff
@@ -176,99 +182,80 @@ export abstract class BinaryRunner {
...
-    public asAnalyzerRequestString(...requests: AnalyzeScanRequest[]): string {
-        return yaml.dump({
-            scans: requests
-        } as AnalyzerRequest);
+    public async run(checkCancel: () => void, ...requests: AnalyzeScanRequest[]): Promise<AnalyzerScanResponse | undefined> {
+        // Prepare and validate
+        if (!this.validateSupported()) {
+            return undefined;
+        }
+        let args: RunArgs = this.createRunArguments(...requests);
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/scanLogic/scanRunners/binaryRunner.ts:195]: I understand your point. As I see it, the only purpose of passing `requests` to the Run function is to pass it along to another function. Why not follow a more conventional approach and replace requests: `AnalyzeScanRequest[] `with args: `RunArgs`?

File: src/main/scanLogic/scanRunners/binaryRunner.ts [hunk 5]
```diff
@@ -289,29 +269,40 @@ export abstract class BinaryRunner {
...
      * Copy a file that includes 'log' in its name from a given folder to the logs folder
-     * @param logsSrcFolder - the directory that the log file exists in
-     * @param requestType - the type of runner request of the log, will be part of it's name
-     * @param rootName - the type of runner request of the log, will be part of it's name
+     * @param arg - the run arguments that related this log
      * @param hadError - if true, will log result as error, otherwise success.
      */
-    private copyRunLogToFolder(logsSrcFolder: string, requestType: AnalyzerType, rootName: string, hadError: boolean) {
-        let logFile: string | undefined = fs.readdirSync(logsSrcFolder).find(fileName => fileName.toLowerCase().includes('log'));
+    private copyRunLogToFolder(args: RunArgs, hadError: boolean) {
```
Comment (Or-Geva) [inline src/main/scanLogic/scanRunners/binaryRunner.ts:288]: Let's add a tests for this copy function

File: src/test/tests/scanAnlayzerRunner.test.ts [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+            shouldFail: false,
+            url: 'platformUrl',
+            user: '',
+            pass: '',
+            token: 'access-token',
+            proxy: 'proxyUrlEnvVarValue',
+            logPath: undefined
+        }
+    ].forEach(test => {
+        it('Create environment variables for binary execution - ' + test.name, () => {
```
Comment (Or-Geva) [inline src/test/tests/scanAnlayzerRunner.test.ts:110]: `it` title should be clear about what is being tested and/or the expected results

**Review by Or-Geva — CHANGES_REQUESTED** [review]

File: src/main/scanLogic/scanRunners/binaryRunner.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (attiasas) [reply src/main/scanLogic/scanRunners/binaryRunner.ts:195]: changed the name of request to args

File: src/main/scanLogic/scanRunners/binaryRunner.ts [hunk 7]
```diff
@@ -113,48 +121,52 @@ export abstract class BinaryRunner {
...
-            }
-            if (proxyHttpsUrl) {
-                binaryVars.HTTPS_PROXY = this.addOptionalProxyAuthInformation(proxyHttpsUrl);
-            }
+    public createEnvForRun(executionLogDirectory?: string): NodeJS.ProcessEnv | undefined {
+        if (!this._connectionManager.areXrayCredentialsSet()) {
+            return undefined;
+        }
+
+        let binaryVars: NodeJS.ProcessEnv = { JFROG_CLI_LOG_LEVEL: Translators.toAnalyzerLogLevel(Configuration.getLogLevel()) };
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/main/scanLogic/scanRunners/binaryRunner.ts:129]: this variable sets the log level for the analyzer manager to be the same as the log level in the vscode extension
image

File: src/main/scanLogic/scanRunners/binaryRunner.ts [hunk 8]
```diff
@@ -289,29 +269,40 @@ export abstract class BinaryRunner {
...
-    private copyRunLogToFolder(logsSrcFolder: string, requestType: AnalyzerType, rootName: string, hadError: boolean) {
-        let logFile: string | undefined = fs.readdirSync(logsSrcFolder).find(fileName => fileName.toLowerCase().includes('log'));
+    private copyRunLogToFolder(args: RunArgs, hadError: boolean) {
+        let logFile: string | undefined = fs.readdirSync(args.directory).find(fileName => fileName.toLowerCase().includes('log'));
         if (!logFile) {
             return;
         }
+
+        let roots: Set<string> = new Set<string>();
+        args.requests.forEach(request => request.rawRequests.forEach(raw => raw.roots.forEach(root => roots.add(Utils.getLastSegment(root)))));
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/main/scanLogic/scanRunners/binaryRunner.ts:295]: changed runArgs to class and moved it inside

File: src/main/scanLogic/scanRunners/binaryRunner.ts [hunk 9]
```diff
@@ -176,102 +188,70 @@ export abstract class BinaryRunner {
...
      */
-    private prepareRequests(args: RunArgs, ...requests: AnalyzeScanRequest[]): AnalyzeScanRequest | undefined {
+    private createRunArguments(...requests: AnalyzeScanRequest[]): RunArgs {
+        let args: RunArgs = {
+            directory: ScanUtils.createTmpDir(),
+            split: false,
+            requests: []
+        } as RunArgs;
+
         let actualRequest: { requests: AnalyzeScanRequest[]; responsePaths: string[] } = { requests: [], responsePaths: [] };
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/main/scanLogic/scanRunners/binaryRunner.ts:218]: removed

**Pull request merged**